### DISCUSSION TITLE
`Psych.load/safe_load` usage

### DISCUSSION THREAD

**Discussion by adam12** [jeremyevans/fixture_dependencies#36] [General]
I'm not sure this is a bug because it's easily worked around, but opening for discussion anyways.

Given a fixture with a plain timestamp (not wrapped in quotes to make it a String).

```
open:
  started_at: 2017-01-01 00:00:00
```

Raises this exception when using newer versions of Psych gem.

```
Psych::DisallowedClass: Tried to load unspecified class: Time
.. elided a little bit of the psych backtrace ..
    /usr/local/lib/ruby/3.1.0/psych.rb:334:in `safe_load'
    /usr/local/lib/ruby/3.1.0/psych.rb:369:in `load'
    /home/app/.gems/gems/fixture_dependencies-1.11.0/lib/fixture_dependencies.rb:116:in `load_yaml'
    /home/app/.gems/gems/fixture_dependencies-1.11.0/lib/fixture_dependencies.rb:177:in `use'
    /home/app/.gems/gems/fixture_dependencies-1.11.0/lib/fixture_dependencies.rb:278:in `block (2 levels) in use'
.. elided a little bit of the fixture_dependencies backtrace ..
```

Switching the value to a String fixes this. As a fix, we could `unsafe_load` instead since presumably all fixtures are ours and we trust them.

Thoughts?

**Comment by jeremyevans** [comment]
Probably easiest to just quote it.  For security reasons, switching to `unsafe_load` seems like a bad idea.  I'll consider a pull request that supports Time/Date classes without switching to `unsafe_load` (assuming that is possible), as long as it is backwards compatible.

**Comment by adam12** [comment]
I'll just quote it for now. Maybe in the new year, I'll look into submitting a PR.

Thanks.
